app/controllers/main/routes.py

```diff
@@ -43,8 +42,10 @@ def redirectToLogout():
 @main_bp.route('/', methods=['GET'])
 def landingPage():
     managerProgramDict = getManagerProgramDict(g.current_user)
-    programsWithEvents = list(ProgramEvent.select(ProgramEvent.program).where(ProgramEvent.event.term == g.current_term).join(Event).distinct())
-    programsWithEventsList = [program.program.id for program in programsWithEvents]
+
+
+    programsWithEvents = list(Event.select(Event.program).where(Event.term == g.current_term).distinct())
```

Comment:
don't limit the select (don't pass any arguments), and you don't need `.distinct()` anymore.

Reply:
The variable also isn't `programsWithEvents`, it's `eventsInTerm` or something.